src/env-inl.h

```diff
@@ -325,6 +326,34 @@ inline void Environment::set_printed_error(bool value) {
   printed_error_ = value;
 }
 
+inline void Environment::print_sync_warn() const {
+  if (!warn_on_sync_)
+    return;
+
+  v8::HandleScope handle_scope(isolate_);
+  v8::Local<v8::StackTrace> stack = v8::StackTrace::CurrentStackTrace(
+      isolate_, 10, v8::StackTrace::kDetailed);
+
+  fprintf(stderr, "WARNING: Detected use of sync API\n");
+  for (int i = 0; i < stack->GetFrameCount() - 1; i++) {
```

Comment:
Yeah. I see how complex the formatting is for the stack trace in `CallSiteToString()` in `deps/v8/src/message.js`.

@bnoordhuis If we were to call to JS, what would be the best way to do that from an env function?